packages/grafana-ui/src/components/DateTimePickers/DateTimePicker/DateTimePicker.tsx

```diff
@@ -0,0 +1,214 @@
+import React, { FC, FormEvent, ReactNode, useCallback, useEffect, useState } from 'react';
+import { useMedia } from 'react-use';
+import Calendar from 'react-calendar/dist/entry.nostyle';
+import { css, cx } from '@emotion/css';
+import { dateTimeFormat, DateTime, dateTime, GrafanaTheme2 } from '@grafana/data';
+import { Button, Field, HorizontalGroup, Icon, Input, Portal } from '../..';
+import { ClickOutsideWrapper } from '../../ClickOutsideWrapper/ClickOutsideWrapper';
+import { isValid } from '../utils';
+import { getBodyStyles, getStyles as getCalendarStyles } from '../TimeRangePicker/TimePickerCalendar';
+import { useStyles2, useTheme2 } from '../../../themes';
+import { TimeOfDayPicker } from '../TimeOfDayPicker';
+
+export interface Props {
+  /** Input date for the component */
+  date?: DateTime;
+  /** Callback for returning the selected date */
+  onChange: (date: DateTime) => void;
+  /** label for the input field */
+  label?: ReactNode;
+}
+
+const stopPropagation = (event: React.MouseEvent<HTMLDivElement>) => event.stopPropagation();
+
+export const DateTimePicker: FC<Props> = ({ date, label, onChange }) => {
+  const [isOpen, setOpen] = useState(false);
+
+  const theme = useTheme2();
+  const isFullscreen = useMedia(`(min-width: ${theme.breakpoints.values.lg}px)`);
+  const containerStyles = useStyles2(getCalendarStyles);
+  const styles = useStyles2(getStyles);
+
+  const onApply = useCallback(
+    (date: DateTime) => {
+      setOpen(false);
+      onChange(date);
+    },
+    [onChange]
+  );
+
+  const onOpen = useCallback(
+    (event: FormEvent<HTMLElement>) => {
+      event.preventDefault();
+      setOpen(true);
+    },
+    [setOpen]
+  );
+
+  return (
+    <div data-testid="date-time-picker" style={{ position: 'relative' }}>
+      <DateTimeInput date={date} onChange={onChange} isFullscreen={isFullscreen} onOpen={onOpen} label={label} />
+      {isOpen ? (
+        isFullscreen ? (
+          <ClickOutsideWrapper onClick={() => setOpen(false)}>
+            <DateTimeCalendar date={date} onChange={onApply} isFullscreen={true} onClose={() => setOpen(false)} />
```

Comment:
Since it's possible to manually modify date via the input field, there should be some validation added for the date prop, if that's not a valid date it can default to the `new Date()`.